api/metrics/views.py

```diff
@@ -69,16 +70,18 @@ def format_response(self, response, query_params):
             'data': data,
         }
 
-    def execute_search(self, search):
-        # TODO - this is copied from get_count_for_preprint in metrics.py - abstract this out in the future
-        try:
-            response = search.execute()
-        except NotFoundError:
-            # _get_relevant_indices returned 1 or more indices
-            # that doesn't exist. Fall back to unoptimized query
-            search = search.index().index(self.metric._default_index())
-            response = search.execute()
-        return response
+    def execute_search(self, query=None, search=None):
+        if search is None:
+            search = self.metric.search()
+
+        es = get_connection(search._using)
+
+        if query:
+            es_search = es.search(index=search._index, body=query)
+        else:
+            es_search = es.search(index=search._index)
+
+        return search._response_class(search, es_search)
```

Comment:
The 'why' for changing all of this would be good for the Changes section of the PR description. Also how it makes it unnecessary to to catch the NotFoundError or if we're losing something by not using the `_get_relevant_indices` method that `search.execute()` apparently called.

Reply:
Would also make for a good doc string here.